### PR TITLE
ENH: Convert finfo.tiny to finfo.smallest_normal

### LINKED ISSUE #70909: `finfo(...).tiny` should be aliased to `finfo(...).smallest_normal`
The [array API specification stipulates](https://data-apis.org/array-api/latest/API_specification/data_type_functions.html?highlight=smallest_normal#finfo-type) that the

> smallest positive floating-point number with full precision

should be called `smallest_normal`. We are currently calling it [`tiny`](https://pytorch.org/docs/master/type_info.html#torch.torch.finfo). See [this discussion](https://github.com/data-apis/array-api/pull/129#discussion_r577993658) for the motivation behind this.

cc @mruberry @rgommers @pmeier @asmeurer @leofang @AnirudhDagar @asi1024 @emcastillo @kmaehashi

### PR DESCRIPTION
**Pull request by HaoZeke**
Fixes #70909, by a straightforward search and replace discussed in #70909.

### CODE REVIEW

**Comment by mruberry** [comment]
The discussion in the issue says they should be aliases, though, not that tiny should be replaced by smallest_normal. We should support both.

**Comment by HaoZeke** [comment]
> The discussion in the issue says they should be aliases, though, not that tiny should be replaced by smallest_normal. We should support both.

@mruberry, yup, sorry about that; does this look alright now?

File: test/test_type_info.py [hunk 1]
```diff
@@ -63,6 +63,7 @@ def test_finfo(self):
         self.assertEqual(xinfo.min, -3.38953e+38)
         self.assertEqual(xinfo.eps, 0.0078125)
         self.assertEqual(xinfo.tiny, 1.17549e-38)
+        self.assertEqual(xinfo.tiny, xinfo.smallest_normal)
```
Comment (mruberry) [inline test/test_type_info.py:66]: Nice

**Review by mruberry — APPROVED** [review]
Cool! Thanks @HaoZeke!

**Pull request closed without merging**